offsets-tracker/cache/cache.go

```diff
@@ -0,0 +1,70 @@
+package cache
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/keyval-dev/offsets-tracker/binary"
+	"github.com/keyval-dev/offsets-tracker/schema"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type Cache struct {
+	data *schema.TrackedOffsets
+}
+
+func NewCache() *Cache {
+	f, err := os.Open(schema.FileName)
+	if err != nil {
+		fmt.Println("could not find existing offset file, cache will be empty")
+		return nil
+	}
+
+	defer f.Close()
+	data, err := ioutil.ReadAll(f)
+	if err != nil {
+		log.Fatalf("error parsing existing offsets file: %v\n", err)
```

Comment:
Is it ok to proceed further in case of error? (should not we return nil in this case) and the same below